### PR TITLE
[Internal][Executor] Update running nodes status to canceled when recieving cancel request in executor server

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Update running nodes status to canceled when recieving cancel request in executor server.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 1]
```diff
@@ -936,12 +936,14 @@ def _exec(
             # KeyboardInterrupt will be raised after asyncio finishes its signal handling
             # End run with the KeyboardInterrupt exception, so that its status will be Canceled
             flow_logger.info("Received KeyboardInterrupt, cancel the run.")
+            # Update the run info of those running nodes to a canceled status.
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:939]: Will a conflict occur here? An async function would also be cancelled in it, right?

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 2]
```diff
@@ -1327,3 +1343,8 @@ def execute_flow(
             # remove line_number from output
             line_result.output.pop(LINE_NUMBER_KEY, None)
         return line_result
+
+
+def signal_handler(sig, frame):
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:1348]: If it is a special signal handling logic, we may add more document to explain the scenario.

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py:1346]: Updatd.

File: src/promptflow/promptflow/executor/flow_executor.py (hunk 1 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py:949]: An async function will be cancelled before this part.

**Review by Hhhilulu — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**